Fix the badIlOp node generated by RegDepCopyRemoval

r30shah opened the pull request:
For creating a fresh copy for RegDepCopyRemoval optimization, there was a case when a regLoad is under PassThrough node in which case we would not have corresponding regStore. To handle this case we were creating a regStore by getting the type from PassThrough node which would cause badIlOp. This commit fixes that issue.

Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

r30shah commented:
@andrewcraik  Can I get the review on this change? There was a failure found on internal testing where it was generating `badILOp` . 
FYI @fjeremic  @dchopra001

andrewcraik commented:
@genie-omr build all

0xdaryl commented:
Minor nit:  can you fix the spelling of "generated" in the commit title please?  CI doesn't need to be run afterwards if that is the only change.

r30shah commented:
Thanks @0xdaryl  for review. I have fixed the typo in the commit and title of this PR. @andrewcraik I have rebased the repo to re-run the eclipse.omr tests and seems like they are passing. We can relaunch the build.

andrewcraik commented:
@genie-omr build all

r30shah commented:
@andrewcraik  All the tests have been passed. This one is good to merge
